### PR TITLE
Refactor usage of str() in providers

### PR DESCRIPTION
**Pull request by eumiro**

### CODE REVIEW

File: tests/providers/qubole/operators/test_qubole_check.py
Comment (Taragolis) [inline tests/providers/qubole/operators/test_qubole_check.py:1]: Suspended provider

File: airflow/providers/common/sql/operators/sql.py [hunk 1]
```diff
@@ -945,8 +945,8 @@ def __init__(
         sqlexp = ", ".join(self.metrics_sorted)
         sqlt = f"SELECT {sqlexp} FROM {table} WHERE {date_filter_column}="
 
-        self.sql1 = sqlt + "'{{ ds }}'"
-        self.sql2 = sqlt + "'{{ macros.ds_add(ds, " + str(self.days_back) + ") }}'"
+        self.sql1 = f"{sqlt}'{{ ds }}'"
```
Comment (Taragolis) [inline airflow/providers/common/sql/operators/sql.py]:
```suggestion
        self.sql1 = f"{sqlt}'{{{{ ds }}}}'"
```

File: airflow/providers/celery/executors/celery_executor_utils.py [hunk 2]
```diff
@@ -155,7 +155,7 @@ def _execute_in_fork(command_to_exec: CommandType, celery_task_id: str | None =
         args.func(args)
         ret = 0
     except Exception as e:
-        log.exception("[%s] Failed to execute task %s.", celery_task_id, str(e))
+        log.exception("[%s] Failed to execute task %s.", celery_task_id, e)
```
Comment (uranusjr) [inline airflow/providers/celery/executors/celery_executor_utils.py]: Probably not belong in this PR but this second argument really isn’t needed since `exception` would log this automatically.

File: airflow/providers/common/sql/operators/sql.py (hunk 1 above)
Comment (uranusjr) [reply airflow/providers/common/sql/operators/sql.py:948]: Or

```suggestion
        self.sql1 = f"{sqlt}" "'{{ ds }}'"
```

File: airflow/providers/google/suite/transfers/local_to_drive.py [hunk 3]
```diff
@@ -119,7 +119,7 @@ def execute(self, context: Context) -> list[str]:
             try:
                 remote_file_id = hook.upload_file(
                     local_location=str(local_path),
-                    remote_location=str(Path(self.drive_folder) / Path(local_path).name),
+                    remote_location=Path(self.drive_folder, local_path).name,
```
Comment (uranusjr) [inline airflow/providers/google/suite/transfers/local_to_drive.py]: This does not seem to be equivalent?

File: airflow/providers/grpc/hooks/grpc.py [hunk 4]
```diff
@@ -83,7 +83,7 @@ def get_conn(self) -> grpc.Channel:
         base_url = self.conn.host
 
         if self.conn.port:
-            base_url = base_url + ":" + str(self.conn.port)
+            base_url += f":{self.conn.port}"
```
Comment (uranusjr) [inline airflow/providers/grpc/hooks/grpc.py:86]:
```suggestion
            base_url = f"{base_url}:{self.conn.port}"
```

Comment (uranusjr) [reply airflow/providers/grpc/hooks/grpc.py:86]: There are other similar `+=` usages that can be improved in this PR.

File: airflow/providers/google/suite/transfers/local_to_drive.py (hunk 3 above)
Comment (eumiro) [reply airflow/providers/google/suite/transfers/local_to_drive.py]: Oh, my bad, thanks!

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 5]
```diff
@@ -576,8 +578,7 @@ async def get_files_async(
             response = paginator.paginate(Bucket=bucket, Prefix=prefix, Delimiter=delimiter)
             async for page in response:
                 if "Contents" in page:
-                    _temp = [k for k in page["Contents"] if isinstance(k.get("Size", None), (int, float))]
-                    keys = keys + _temp
+                    keys.extend(k for k in page["Contents"] if isinstance(k.get("Size", None), (int, float)))
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/s3.py]:
```suggestion
                    keys.extend(k for k in page["Contents"] if isinstance(k.get("Size"), (int, float)))
```

File: airflow/providers/cncf/kubernetes/utils/pod_manager.py [hunk 6]
```diff
@@ -728,23 +728,23 @@ def extract_xcom_kill(self, pod: V1Pod):
...
                 while resp.peek_stderr():
-                    error_res = error_res + resp.read_stderr() if error_res else resp.read_stderr()
+                    error_res += resp.read_stderr()
                 if error_res:
                     self.log.info("stderr from command: %s", error_res)
                     break
                 if res:
                     return res
-        return res
+        return None
```
Comment (uranusjr) [inline airflow/providers/cncf/kubernetes/utils/pod_manager.py:748]: This function can benefit from a rewrite using early return to remove a level of indentation.

**Review by uranusjr — APPROVED** [review]

File: airflow/providers/cncf/kubernetes/utils/pod_manager.py [hunk 7]
```diff
@@ -728,23 +728,24 @@ def extract_xcom_kill(self, pod: V1Pod):
...
-                    self.log.info("stderr from command: %s", error_res)
-                    break
-                if res:
-                    return res
-        return res
+        res = ""
+        if not resp.is_open():
+            return None
+        self.log.info("Running command... %s", command)
+        resp.write_stdin(command + "\n")
```
Comment (uranusjr) [inline airflow/providers/cncf/kubernetes/utils/pod_manager.py]:
```suggestion
        resp.write_stdin(f"{command}\n")
```

or

```suggestion
        resp.write_stdin(command)
        resp.write_stdin("\n")
```

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]
